### PR TITLE
[FLINK-23559][tests] Randomize periodic materialisation interval in tests

### PR DESCRIPTION
**Pull request by rkhachatryan**

### CODE REVIEW

File: flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java
Comment (curcur) [inline flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java:141]: This is a good idea.

But do we have ITTests that last more than 10 minutes?

If not I think the last two are equivalent saying that materialization is not played?

**Review by curcur — APPROVED** [review]

**Comment by curcur** [comment]
I've accepted the PR with some comments.

You can decide with your own judgement whether to keep the last two or not.

File: flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java
Comment (rkhachatryan) [reply flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java:141]: I think you're right, 10 minutes is probably too much, I'll remove it.

**Comment by rkhachatryan** [comment]
Thanks for the review!
I removed the 10-minute option and rebased on the latest master (both required PRs are merged).
Will merge after passing the tests.

**Comment by rkhachatryan** [comment]
Rebased to included the latests fixes, waiting for CI to merge.

**Comment by rkhachatryan** [comment]
Private build passed except `e2e_2_ci_build` phase which passed in PR build, merging.

**Pull request merged**
